guillotina/api/container.py

```diff
@@ -114,8 +114,12 @@ class DefaultPOST(Service):
         if "@type" not in data or data["@type"] not in app_settings["container_types"]:
             raise HTTPNotFound(content={"message": "can not create this type %s" % data["@type"]})
 
-        if "id" not in data:
-            raise HTTPPreconditionFailed(content={"message": "We need an id"})
+        try:
+            assert data["id"]
```

Comment:
I think it's better to change the assert for an if, because assert can be disabled when running Python with `-O` (https://stackoverflow.com/questions/1273211/disable-assertions-in-python)

Reply:
Also, maybe the fix should be applied here https://github.com/plone/guillotina/blob/73a4793e70531105a2560863e818b758823dc3f1/guillotina/content.py#L561 ? Is it intended make ID optional?